File: 卡码/代码随想录/1/08-贪心算法/Java/greedy/src/main/java/com/kurumi/greedy/_02_中等/LeetCode_0045.java
//45. 跳跃游戏 II
package com.kurumi.greedy._02_中等;

public class LeetCode_0045 {

}
class Solution {
  public int jump(int[] nums) {
    int cur=0,next=0,res=0;
    for(int i=0;i<nums.length-1;i++){
      next=Math.max(i+nums[i], next);
      if(cur==i){
        cur=next;
        res++;
      }
    }
    return res;
  }
}

File: 卡码/代码随想录/1/08-贪心算法/Python/02-中等/LeetCode_0045.py
#45. 跳跃游戏 II
from typing import *

class Solution:
  def jump(self, nums: List[int]) -> int:
    res,cur,next=0,0,0
    for i in range(len(nums)-1):
      next=max(next,i+nums[i])
      if i==cur:
        cur=next
        res+=1
    return res